File: app/src/main/java/com/shuoye/video/utils/network/converter/FastJsonRequestBodyConverter.java
package com.shuoye.video.utils.network.converter;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.serializer.SerializeConfig;
import okhttp3.MediaType;
import okhttp3.RequestBody;
import org.jetbrains.annotations.NotNull;
import retrofit2.Converter;

final class FastJsonRequestBodyConverter<T> implements Converter<T, RequestBody> {

    private static final MediaType MEDIA_TYPE = MediaType.parse("application/json; charset=UTF-8");

    private final SerializeConfig serializeConfig;

    public FastJsonRequestBodyConverter(SerializeConfig serializeConfig) {
        this.serializeConfig = serializeConfig;
    }

    @Override
    public RequestBody convert(@NotNull T value) {
        return RequestBody.create(MEDIA_TYPE, JSON.toJSONBytes(value, serializeConfig));
    }
}

File: app/src/main/java/com/shuoye/video/api/converter/FastJsonRequestBodyConverter.kt
package com.shuoye.video.api.converter

import com.alibaba.fastjson.JSON
import com.alibaba.fastjson.serializer.SerializeConfig
import okhttp3.MediaType
import okhttp3.MediaType.Companion.toMediaTypeOrNull
import okhttp3.RequestBody
import okhttp3.RequestBody.Companion.toRequestBody
import retrofit2.Converter


/**
 * 转换服务器返回数据
 *
 * @author shuoye
 */
internal class FastJsonRequestBodyConverter<T>(
    private val serializeConfig: SerializeConfig
) : Converter<T, RequestBody> {

    override fun convert(value: T): RequestBody {
        return JSON.toJSONBytes(value, serializeConfig).toRequestBody(MEDIA_TYPE, 0)
    }

    companion object {
        private val MEDIA_TYPE: MediaType? = "application/json; charset=UTF-8".toMediaTypeOrNull()
    }
}